WordPress/Classes/ViewRelated/Post/Scheduling/CalendarCollectionView.swift

```diff
@@ -0,0 +1,134 @@
+
+        scrollDirection = .horizontal
+        scrollingMode = .stopAtEachCalendarFrame
+        showsHorizontalScrollIndicator = false
+
+        ibCalendarDataSource = calDataSource
+        ibCalendarDelegate = calDataSource
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        register(DateCell.self, forCellWithReuseIdentifier: "dateCell")
+
+        scrollDirection = .horizontal
+        scrollingMode = .stopAtEachCalendarFrame
+        showsHorizontalScrollIndicator = false
+
+        ibCalendarDataSource = calDataSource
+        ibCalendarDelegate = calDataSource
+    }
+}
+
+class CalendarDataSource: JTACMonthViewDataSource {
+
+    var willScroll: ((DateSegmentInfo) -> Void)?
+    var didScroll: ((DateSegmentInfo) -> Void)?
+    var didSelect: ((Date) -> Void)?
+
+    func configureCalendar(_ calendar: JTACMonthView) -> ConfigurationParameters {
+        let startDate = Date.farPastDate
+        let endDate = Date.farFutureDate
+        return ConfigurationParameters(startDate: startDate, endDate: endDate)
+    }
+}
+
+extension CalendarDataSource: JTACMonthViewDelegate {
+    func calendar(_ calendar: JTACMonthView, cellForItemAt date: Date, cellState: CellState, indexPath: IndexPath) -> JTACDayCell {
+        let cell = calendar.dequeueReusableJTAppleCell(withReuseIdentifier: "dateCell", for: indexPath) as! DateCell
+        configureCell(view: cell, cellState: cellState)
+        return cell
+    }
+
+    func calendar(_ calendar: JTACMonthView, willDisplay cell: JTACDayCell, forItemAt date: Date, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+    }
+
+    func calendar(_ calendar: JTACMonthView, willScrollToDateSegmentWith visibleDates: DateSegmentInfo) {
+        willScroll?(visibleDates)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didScrollToDateSegmentWith visibleDates: DateSegmentInfo) {
+        didScroll?(visibleDates)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didSelectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+        didSelect?(date)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didDeselectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+    }
+
+    private func configureCell(view: JTACDayCell?, cellState: CellState) {
+       guard let cell = view as? DateCell else { return }
+       cell.dateLabel?.text = cellState.text
+       handleCellTextColor(cell: cell, cellState: cellState)
+    }
+
+    private func handleCellTextColor(cell: DateCell, cellState: CellState) {
+        if cellState.dateBelongsTo == .thisMonth {
+          cell.dateLabel?.textColor = .text
+        } else {
+          cell.dateLabel?.textColor = .textSubtle
+        }
+
+        if cellState.isSelected {
```

Comment:
Good catch, probably shouldn't override the previously set color. Can't do the latter because the inverted text color overrides both branches, but I can do the following:

```swift
if cellState.isSelected {
  cell.dateLabel?.textColor = .textInverted
} else if cellState.dateBelongsTo == .thisMonth {
  cell.dateLabel?.textColor = .text
} else {
  cell.dateLabel?.textColor = .textSubtle
}
```